src/System.IO.Pipelines/MemoryEnumerator.cs

```diff
@@ -17,7 +17,7 @@ public struct MemoryEnumerator
         public MemoryEnumerator(ReadCursor start, ReadCursor end)
         {
             _segmentEnumerator = new SegmentEnumerator(start, end);
-            _current = Memory<byte>.Empty;
+            _current = default(Memory<byte>);
```

Comment:
I will make Empty faster. The current implementation is pretty naive. It calls OwnedMemory.Memory every time get_Empty is called.